rust/src/smtp/smtp.rs

```diff
@@ -0,0 +1,165 @@
+/* Copyright (C) 2021 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+use crate::core::Direction;
+
+pub struct RustSMTPBuffer {
+    ts_buf: Vec<u8>,
+    tc_buf: Vec<u8>,
+    cur_line: Vec<u8>,
+}
+
+impl RustSMTPBuffer {
+
+    pub fn new() -> Self {
+        return Self {
+            ts_buf: Vec::new(),
+            tc_buf: Vec::new(),
+            cur_line: Vec::new(),
+        }
+    }
+
+    pub fn get_line_ts(&mut self, input: &[u8]) -> i8
```

Comment:
so instead of the `-> i8` do something like `-> Option(Vec<u8>)` here? Or maybe a slice instead of a Vec

Reply:
Probably a `Vec<u8>` as we can split off the line from `ts_buf` or `tc_buf`